dungeon/src/quizquestion/DummyQuizQuestionList.java

```diff
@@ -110,15 +108,14 @@ public class DummyQuizQuestionList {
                             new QuizQuestion(
                                     new QuizQuestionContent(
                                             QuizQuestionContent.QuizQuestionContentType.IMAGE,
-                                            Constants.EMPTY_MESSAGE),
+                                            "image_quiz/dummy.png"),
```

Comment:
magic strings mitten im code?

Reply:
Ich würde hier sagen ist erlaubt da es sich um eine Sammlung von Vorlage Fragen sind welche vielleicht in einem neuen Issue von einer Config oder so eingelesen werden und nicht mehr im Code. Glaube, die DummyQuizQuestionList wurde erstellt, um überhaupt Fragen zu haben, welche genutzt werden können zum Testen. 

Hier ist es auch etwas "komisch", da die Liste wo die Fragen hereingeworfen sind, ja bereits "static final" ist.

Vielleicht hier einfach mal nur zur Sicherheit soll ich die DummyQuizQuestionList einmal überarbeiten, oder nicht? Wenn ja direkt hier oder in neuen Issue?

@cagix @AMatutat pinge euch hierfür einmal an

Reply:
Aus meiner Sicht können da Magic-Strings drinnen bleiben. 
Das sind konkrete Dummy-Fragen zum Testen und langfristig gesehen wird das durch DSL-Eingaben ersetzt. 
Aus dem Grund sehe ich jetzt auch keinen aktiven Bedarf das Ding nochmal in irgendeiner Form zu refactorn.

Reply:
Ja, habe den Kontext nicht gesehen und nur die geänderte Code-Stelle. Habe grad die ursprünglichen (ausgeblendeten) Code-Stellen aufgemacht (`new QuizQuestion(...new QuizQuestionContent(...new QuizQuestionContent[]...new QuizQuestionContent(...)`) und fast nen Herzkasper bekommen. **=> Ignoriert meinen Einwurf bzgl. der Magic Strings.**

Reply:
> Das sind konkrete Dummy-Fragen zum Testen

@AHeinisch @AMatutat Öhm, dumme Frage: Warum taucht das dann in der Code-Basis auf? Sollte das dann nicht unter `tests/...` stehen?

Reply:
> @AHeinisch @AMatutat Öhm, dumme Frage: Warum taucht das dann in der Code-Basis (`src/`) auf? Sollte das dann nicht unter `test/...` stehen?

Testen der UI mit unterschiedlichen Fragen. Nicht testen mithilfe von JUnit. Leider handelt es sich hier um einen grafischen Test, dass theoretisch die Texte und Bilder angezeigt werden können und nicht außerhalb des dargestellten Bereiches.

Reply:
ja, aber trotzdem nicht produktivcode/`src/` :) 

in `test/` können ja auch manuelle tests stehen. brauchst ja nur das `@Test` weglassen...

Reply:
Puh, ich weiß nicht...
Ich finde beides unschön, aber irgendwie gehören in test/ für mich nur Testfälle.
Besonders in diesem konkreten Fall, wo ich im richtigen Produktivcode (z.B. in `Game`) dann auf meinen Test-/Dummy-Code zugreifen muss. Dann importiere ich mir Sachen aus dem test/-Ordner... Klingt richtig verkehrt.

Reply:
> Besonders in diesem konkreten Fall, wo ich im richtigen Produktivcode (z.B. in `Game`) dann auf meinen Test-/Dummy-Code zugreifen muss. Dann importiere ich mir Sachen aus dem test/-Ordner... Klingt richtig verkehrt.

Ja, das geht tatsächlich nicht. Ich finde es aber trotzdem unschön, Dummy-Code im Produktivcode drin zu haben. 

Nochmal einen Schritt zurück: Was ist denn das Ziel bzw. der Zweck von diesem ganzen Dummy-Zeugs? Geht es nur darum zu schauen, ob irgendwas genau so zusammengesteckt werden kann? Dann macht daraus einen Testfall, der mit JUnit kompiliert (aber nicht wirklich ausgeführt) wird. Geht es darum, dem potentiellen User irgendwas zu demonstrieren? Dann gehört das in einem Hilfsbranch ausprobiert (lauffähig gemacht) und als Code-Beispiel in die Doku kopiert (aber nicht als Code in den `master` eingecheckt).

Reply:
Ja (zu allem),
Es dient uns aber auch als Placeholder, um das "Backend" weiterzuentwickeln solange die DSL-Andindung noch nicht steht.

Lasst uns diese Diskussion mal in #661 verlagern.
Ich glaube mit #688 haben wir auch eine bessere möglichkeit diesen Placeholder/Test/Dummy in das gesamtkonstrukt zu integrieren.

Reply:
done